src/main/java/biz/netcentric/filevault/validator/aem/cloud/AemCloudValidator.java

```diff
@@ -66,36 +75,44 @@ public AemCloudValidator(boolean allowVarNodesOutsideContainers, boolean allowLi
         this.defaultSeverity = defaultSeverity;
         this.foundViolation = false;
         this.hasMutableNodes = false;
+        this.hasImmutableNodes = false;
         this.hasInstallHooks = false;
     }
 
     @Override
     public Collection<ValidationMessage> validate(@NotNull String path) {
-        if (!foundViolation && path.startsWith("/var/")) {
+        Collection<ValidationMessage> messages = new ArrayList<>();
+        if (numVarNodeViolations < MAX_NUM_VIOLATIONS_PER_TYPE && path.startsWith("/var/")) {
             // check if package itself is only used on author
             if (!allowVarNodesOutsideContainers || !isContainedInAuthorOnlyPackage(containerValidationContext)) {
                 // only emit once per package
                 foundViolation = true;
-                return Collections.singleton(new ValidationMessage(defaultSeverity, String.format(
-                        VIOLATION_MESSAGE_VAR_NODES, allowVarNodesOutsideContainers ? VIOLATION_MESSAGE_VAR_NODES_CONDITION_CONTAINER : VIOLATION_MESSAGE_VAR_NODES_CONDITION_OVERALL)));
+                messages.add(new ValidationMessage(defaultSeverity, String.format(
+                        VIOLATION_MESSAGE_VAR_NODES, allowVarNodesOutsideContainers ? VIOLATION_MESSAGE_VAR_NODES_CONDITION_CONTAINER
+                                : VIOLATION_MESSAGE_VAR_NODES_CONDITION_OVERALL)));
+                numVarNodeViolations++;
             }
         }
-        if (!path.startsWith("/apps/") && !path.startsWith("/libs")) {
+        if (!path.startsWith("/apps/") && !path.startsWith("/libs/") && !path.startsWith("/oak:index/")) {
```

Comment:
The paths `/` (root), `/apps`, `/libs` and `/oak:index` (without the trailing `/`) will enter in this `if` block and will be considered as mutable content for this validator.

Reply:
Good observation, it was working in my tests somehow but this method is called for both folders and files, so the check needs to be extended.

Reply:
Improved the check in https://github.com/Netcentric/aem-cloud-validator/pull/13/commits/060486e7100663013db2d79aabeeb37cbd4d8def.

Reply:
The new implementation fixes all those cases except for the root path (`/`). I am still getting this error for a mixed package:
`[ERROR] ValidationViolation: "netcentric-aem-cloud: Mutable nodes in mixed package types are not installed!", filePath=jcr_root, nodePath=/`

To fix it, I think it is better not to consider `/` neither as mutable nor as immutable, but to skip it earlier.

That way we can also avoid setting the variables `hasMutableNodes` or `hasImmutableNodes` to `true` for `/`, and therefore reporting false positives in the validation in the method `done()` - `if (packageType == null && hasMutableNodes && hasImmutableNodes)`. For example, with the current implementation:
- If `/` is considered mutable, and we have also path like `/apps/myapp/content.xml`, that validation would fail because the latter is immutable.
- If `/` is considered immutable, and we have also path like `/content/myapp/content.xml`, that validation would fail because the latter is mutable.

Reply:
Right, should be fixed with https://github.com/Netcentric/aem-cloud-validator/pull/13/commits/72f317d2f5d2291fb2925ee2f9a16e1cffbe0bea.